x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -125,45 +145,152 @@ export class ReportingStore {
         browser_type: report.browser_type,
       },
     };
-    return this.client.callAsInternalUser('index', indexParams);
+    return await this.client.callAsInternalUser('index', indexParams);
   }
 
+  /*
+   * Called from addReport, which handles any errors
+   */
   private async refreshIndex(index: string) {
-    return this.client.callAsInternalUser('indices.refresh', { index });
+    return await this.client.callAsInternalUser('indices.refresh', { index });
   }
 
-  public async addReport(type: string, payload: unknown, options: AddReportOpts): Promise<Report> {
+  public async addReport(type: string, username: string | null, payload: unknown): Promise<Report> {
     const timestamp = indexTimestamp(this.indexInterval);
     const index = `${this.indexPrefix}-${timestamp}`;
     await this.createIndex(index);
 
     const report = new Report({
-      index,
+      _index: index,
       payload,
       jobtype: type,
-      created_by: options.created_by,
-      browser_type: options.browser_type,
-      max_attempts: options.max_attempts,
-      timeout: options.timeout,
-      priority: 10, // unused
+      created_by: username,
+      ...this.jobSettings,
     });
 
-    const doc = await this.saveReport(report);
-    report.updateWithDoc(doc);
+    try {
+      const doc = await this.indexReport(report);
+      report.updateWithEsDoc(doc);
 
-    await this.refreshIndex(index);
-    this.logger.info(`Successfully queued pending job: ${report._index}/${report.id}`);
+      await this.refreshIndex(index);
+      this.logger.info(`Successfully stored pending job: ${report._index}/${report._id}`);
 
-    return report;
+      return report;
+    } catch (err) {
+      this.logger.error(`Error in addReport!`);
+      this.logger.error(err);
+      throw err;
+    }
   }
 
-  public async updateReport(query: UpdateQuery): Promise<Report> {
```

Comment:
I split this method up into:
 - setReportClaimed
 - setReportFailed
 - setReportCompleted

to make things easier to test, defining the new status to write within the function instead of taking it as a param.